lib/tools.js

```diff
@@ -1932,6 +1932,63 @@ function removeIdFromAllEnums(objects, id) {
     });
 }
 
+/**
+ * Validates types of obj.common properties and object.type, if invalid types are used, an error is thrown.
+ * If attributes of obj.common are not provided, no error is thrown. obj.type has to be there and has to be valid.
+ *
+ * @param {object} obj an object which will be validated
+ * @throws Error if a property has the wrong type or obj.type is non existing
+ */
+function validateGeneralObjectProperties(obj) {
+    if (!obj || obj.type === undefined) {
+        throw new Error(`obj.type has to exist`);
+    }
+
+    if (typeof obj.type !== 'string') {
+        throw new Error(`obj.type has an invalid type! Expected "string", received  "${typeof obj.type}"`);
+    }
+
+    const allowedObjectTypes = ['state', 'channel', 'device', 'enum', 'host', 'adapter', 'instance', 'meta', 'config', 'script', 'user', 'group', 'chart'];
```

Comment:
type "folder" is missing ... it is kind of "if device and channel is noth (clear) enough and you need an additional level"

Reply:
Maybe also missing in the docs?

Reply:
It seems so.. I took the information from https://github.com/ioBroker/ioBroker.docs/blob/master/docs/en/dev/objectsschema.md#object-types

Reply:
> if device and channel is noth (clear) enough and you need an additional level

I love undocumented features :imp: . I have that exact case in my tradfri adapter. "Solved" it by not using an object at all for the additional level.

Reply:
*gg*

Reply:
"Folder" type was added https://github.com/ioBroker/ioBroker.docs/blob/master/docs/en/dev/objectsschema.md#object-types